hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -342,7 +343,7 @@ public TargetTagAssignmentResult toggleTagAssignment(final Collection<String> ta
 
         // all are already assigned -> unassign
         if (alreadyAssignedTargets.size() == allTargets.size()) {
-            alreadyAssignedTargets.forEach(target -> target.getTags().remove(tag));
+            alreadyAssignedTargets.forEach(target -> ((JpaTarget) target).removeTag(tag));
```

Comment:
If you change the return type of the query `org.eclipse.hawkbit.repository.jpa.TargetRepository.findByTagNameAndControllerIdIn(String, Collection<String>)`  from `Target` to `JpaTarget` you don't need to cast here.